compilation/src/main/scala/scala/tools/nsc/ScalacBenchmark.scala

```diff
@@ -26,53 +37,20 @@ class ScalacBenchmark {
   @Param(value = Array("latest"))
   var corpusVersion: String = _
 
-  var driver: Driver = _
-
   var depsClasspath: String = _
 
-  def compileImpl(): Unit = {
-    val compilerArgs =
-      if (source.startsWith("@")) Array(source)
-      else {
-        import scala.collection.JavaConverters._
-        val allFiles = Files.walk(findSourceDir, FileVisitOption.FOLLOW_LINKS).collect(Collectors.toList[Path]).asScala.toList
-        allFiles.filter(f => {
-          val name = f.getFileName.toString
-          name.endsWith(".scala") || name.endsWith(".java")
-        }).map(_.toAbsolutePath.normalize.toString).toArray
-      }
-
-    // MainClass is copy-pasted from compiler for source compatibility with 2.10.x - 2.13.x
-    class MainClass extends Driver with EvalLoop {
-      def resident(compiler: Global): Unit = loop { line =>
-        val command = new CompilerCommand(line split "\\s+" toList, new Settings(scalacError))
-        compiler.reporter.reset()
-        new compiler.Run() compile command.files
-      }
-
-      override def newCompiler(): Global = Global(settings, reporter)
-
-      override protected def processSettingsHook(): Boolean = {
-        if (source == "scala")
-          settings.sourcepath.value = Paths.get(s"../corpus/$source/$corpusVersion/library").toAbsolutePath.normalize.toString
-        else
-          settings.usejavacp.value = true
-        settings.outdir.value = tempDir.getAbsolutePath
-        settings.nowarn.value = true
-        if (depsClasspath != null)
-          settings.processArgumentString(s"-cp $depsClasspath")
-        if (extraArgs != null && extraArgs != "")
-          settings.processArgumentString(extraArgs)
-        true
-      }
+  def compilerArgs =
+    if (source.startsWith("@")) Array(source)
```

Comment:
Note that `dotc @foo.args` is not yet supported, see https://github.com/lampepfl/dotty/issues/2759